python-checks/src/main/java/org/sonar/python/checks/django/DjangoModelStringFieldCheck.java

```diff
@@ -0,0 +1,142 @@
+    "django.db.models.CharField",
+    "django.db.models.TextField"
+  );
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.CLASSDEF, ctx -> {
+      var classDef = (ClassDef) ctx.syntaxNode();
+      if (TreeUtils.getParentClassesFQN(classDef).contains(DJANGO_MODEL_FQN)) {
+        classDef.body().statements().stream()
+          .filter(AssignmentStatement.class::isInstance)
+          .map(AssignmentStatement.class::cast)
+          .map(AssignmentStatement::assignedValue)
+          .filter(CallExpression.class::isInstance)
+          .map(CallExpression.class::cast)
+          .filter(DjangoModelStringFieldCheck::isTextField)
+          .forEach(call -> validateTextFieldArguments(ctx, call));
+      }
+    });
+  }
+
+  private static boolean isTextField(CallExpression call) {
+    return Optional.of(call)
+      .map(CallExpression::calleeSymbol)
+      .map(Symbol::fullyQualifiedName)
+      .filter(FIELD_TYPES_FQN::contains)
+      .isPresent();
+  }
+
+  private static void validateTextFieldArguments(SubscriptionContext ctx, CallExpression call) {
+    var nullArg = getCallArgumentByName(call, "null")
+      .filter(DjangoModelStringFieldCheck::isArgumentSetAsTrue);
+    var blankArg = getCallArgumentByName(call, "blank")
+      .filter(DjangoModelStringFieldCheck::isArgumentSetAsTrue);
+
+    nullArg.ifPresent(arg -> {
+      if (blankArg.isPresent()) {
+        var issue = ctx.addIssue(arg, REMOVE_MESSAGE);
+        createRemoveArgQuickFix(call, arg).ifPresent(issue::addQuickFix);
+      } else {
+        ctx.addIssue(arg, REPLACE_MESSAGE)
+          .addQuickFix(PythonQuickFix.newQuickFix(REPLACE_QUICK_FIX_MESSAGE,
+            TextEditUtils.replace(arg, "blank=True")));
+      }
+    });
+  }
+
+  private static Optional<RegularArgument> getCallArgumentByName(CallExpression call, String name) {
+    return call.arguments().stream()
+      .filter(RegularArgument.class::isInstance)
+      .map(RegularArgument.class::cast)
+      .filter(arg -> Objects.nonNull(arg.keywordArgument()))
+      .filter(arg -> Optional.of(arg)
+          .map(RegularArgument::keywordArgument)
+          .map(Name::name)
+          .filter(name::equals)
+          .isPresent())
+      .findFirst();
+  }
+
+  private static boolean isArgumentSetAsTrue(RegularArgument arg) {
+    return Optional.of(arg)
+      .map(RegularArgument::expression)
+      .filter(TreeUtils::isBooleanLiteral)
+      .filter(Name.class::isInstance)
+      .map(Name.class::cast)
+      .map(Name::name)
+      .filter("True"::equals)
+      .isPresent();
+  }
+
+  private static Optional<PythonQuickFix> createRemoveArgQuickFix(CallExpression call, Tree arg) {
+    return Optional.ofNullable(call.argumentList())
+      .map(Tree::children)
+      .map(args -> {
+        var count = args.size();
+        var index = args.indexOf(arg);
+        var removeFrom = arg;
+        Tree removeTo;
+        if (index == 0) {
```

Comment:
I don't like too much when we do this kind of index arithmetic. If we can't simplify, maybe it's good to have at least some comments that clarify what we're doing in each branch.

Reply:
done

Reply:
Thanks!